cmdstanpy/stanfit.py

```diff
@@ -1288,87 +1295,462 @@ def __repr__(self) -> str:
+            Default value is ``False``.
+        """
+        if not XARRAY_INSTALLED:
+            raise RuntimeError(
+                "xarray is not installed, cannot produce draws array"
+            )
+        mcmc_vars_list = []
+        dup_vars = []
+        if vars is not None:
+            if isinstance(vars, str):
+                vars_list = [vars]
+            else:
+                vars_list = vars
+            for var in vars_list:
+                if var not in self.metadata.stan_vars_cols:
+                    if inc_sample and var in self.mcmc_sample.stan_vars_cols:
+                        mcmc_vars_list.append(var)
+                        dup_vars.append(var)
+                    else:
+                        raise ValueError('unknown variable: {}'.format(var))
+        else:
+            vars_list = list(self.metadata.stan_vars_cols.keys())
+            if inc_sample:
+                for var in self.mcmc_sample.metadata.stan_vars_cols.keys():
+                    if var not in vars_list and var not in mcmc_vars_list:
+                        mcmc_vars_list.append(var)
+        for var in dup_vars:
+            vars_list.remove(var)
+
+        self._assemble_generated_quantities()
+
+        num_draws = self.mcmc_sample.num_draws_sampling
+        sample_config = self.mcmc_sample.metadata.cmdstan_config
+        attrs = {
+            "stan_version": f"{sample_config['stan_version_major']}."
+            f"{sample_config['stan_version_minor']}."
+            f"{sample_config['stan_version_patch']}",
+            "model": sample_config["model"],
+            "num_unconstrained_params":
+            self.mcmc_sample.num_unconstrained_params,
+            "num_draws_sampling": num_draws,
+        }
+        if inc_warmup and sample_config['save_warmup']:
+            num_draws += self.mcmc_sample.num_draws_warmup
+            attrs["num_draws_warmup"] = self.mcmc_sample.num_draws_warmup
+
+        data = {}
+        coordinates = {"chain": self.chain_ids, "draw": np.arange(num_draws)}
+
+        for var in vars_list:
+            build_xarray_data(
+                data,
+                var,
+                self._metadata.stan_vars_dims[var],
+                self._metadata.stan_vars_cols[var],
+                0,
+                self.draws(inc_warmup=inc_warmup),
+            )
+        if inc_sample:
+            for var in mcmc_vars_list:
+                build_xarray_data(
+                    data,
+                    var,
+                    self.mcmc_sample.metadata.stan_vars_dims[var],
+                    self.mcmc_sample.metadata.stan_vars_cols[var],
+                    0,
+                    self.mcmc_sample.draws(inc_warmup=inc_warmup),
+                )
+
+        return xr.Dataset(data, coords=coordinates, attrs=attrs).transpose(
+            'chain', 'draw', ...
+        )
+
+    def stan_variable(self, name: str, inc_warmup: bool = False) -> np.ndarray:
```

Comment:
Are we okay with `name` as a param or would we prefer `var` to go with the changes to the _pd functions etc?

Also, I assume we will want another (small) pr to change/depreciate the parameter names in MCMC et al's draws functions?

Reply:
did this as part of this PR.